### PR TITLE
Update thermostat.rst

### PR DESCRIPTION
**Pull request by gvdhoven**
Explained "default_mode" in more detail.

## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by kbx81** [comment]
The examples we provide at the top are usually intended to illustrate the minimum configuration to get the component working -- while I don't feel terribly strongly about this, I think we should keep them that way in the interest of streamlining the documentation. If we added a line there every time someone didn't understand one of the options, it would end up looking like what I have in my [config](https://github.com/kbx81/esphome-configs/blob/62ad8ff9d2ad4e69b03b16aabeda80ac4ae8baa2/shared/esp_thermostat_nextion_common_config.yaml#L223-L317); I think it best we avoid that. 🙂 Also, based on what I know about it, I think it's also a safe assumption that most folks will not want the default mode to be a mode where the system is active.

I think the call-out you added is good and adds some clarity; I've got a couple suggestions in terms of wording to make it more consistent with the language/terms we use in the community. I'll leave an in-line comment or two in a moment.

File: components/climate/thermostat.rst [hunk 1]
```diff
@@ -52,6 +52,7 @@ modes that Home Assistant offers.
       - platform: thermostat
         name: "Thermostat Climate Controller"
         sensor: my_temperature_sensor
+        default_mode: heat_cool
```
Comment (kbx81) [inline components/climate/thermostat.rst]: I think we should strike this and keep the examples as simple/minimal as possible.

File: components/climate/thermostat.rst [hunk 2]
```diff
@@ -74,6 +75,7 @@ modes that Home Assistant offers.
       - platform: thermostat
         name: "Thermostat Climate Controller"
         sensor: my_temperature_sensor
+        default_mode: heat
```
Comment (kbx81) [inline components/climate/thermostat.rst]: Same as above

File: components/climate/thermostat.rst [hunk 3]
```diff
@@ -90,6 +92,7 @@ modes that Home Assistant offers.
       - platform: thermostat
         name: "Thermostat Climate Controller"
         sensor: my_temperature_sensor
+        default_mode: cool
```
Comment (kbx81) [inline components/climate/thermostat.rst]: Same as above

File: components/climate/thermostat.rst [hunk 4]
```diff
@@ -173,6 +176,10 @@ These configuration items determine default values the thermostat controller sho
   - ``dry``
   - ``fan_only``
   - ``auto``
+  
+  This value is used if you flash your device for the first time. This means that if you want
+  your thermostat configuration to start in a certain state, you should configure this. If
+  not configured, you have to change the mode (once) via e.g. the UI of Home Assistant.
```
Comment (kbx81) [inline components/climate/thermostat.rst]: Let's re-word this so it's more consistent with the terms/phrasing we use in the community. Here's a suggestion:

```suggestion
This value is used the first time your device starts after ESPHome is initially installed onto it. Add
this option into your configuration if you want your thermostat component to start in a climate mode other
than ``off``.  If this option is not configured, you'll need to manually change the climate mode later via
the front end (Home Assistant), an ESPHome action, automation, or from within a lambda elsewhere in your
device's configuration.
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
Some suggestions, as promised. 😄

**Comment by gvdhoven** [comment]
All comments are resolved.

**Review by kbx81 — APPROVED** [review]
LGTM -- thanks!

**Pull request merged**